[docs] Add more examples to sui client (#17626)
MystenLabs/sui: docs/content/references/cli/client.mdx

╰─────────────┴───────────────────────────────────────────────────────────────────────────────────────────╯
```

### Send SUI or objects

> In this example, let's see how to transfer SUI or transfer an object from one address to another. First all, there two main commands for sending SUI or transfering objects: Pay and Transfer. 

Reviewer: ```suggestion
In this example, let's see how to transfer SUI or transfer an object from one address to another. First of all, there are two main commands for sending SUI or transferring objects: Pay and Transfer. 
```

If "Pay" and "Transfer" are referring to commands, let's use `pay` and `transfer`.

> Each of these commands, has some sister commands: PaySui, PayAllSui, and TransferSui. The main difference between Pay and Transfer is that Pay allows you to send coin to multiple recipients, whereas Transfer only accepts one recipient. Another important difference is that Pay and Transfer require a gas object that is different than the one used to send the coins, which is used to pay gas fees. 

Reviewer: Ditto on formatting of command names -- let's use the form that's recognisable from the CLI (e.g. `pay-sui`, `pay-all-sui` etc). Also, I would describe the differences between these commands as follows:

- Commands that end in `-sui` deal with Sui's native coin, and they use the input coins to pay for gas and for transferring.
- `pay-` commands typically deal with coins and handle smashing for you, whereas `transfer-` commands can handle the transfer of any object that has public transfer (has the `store` ability).

Some special cases:
- `pay-all-sui` is a special case of `pay-sui` that offers a way to transfer the entire balance after smashing.
- `transfer-sui` is a legacy command -- it has been entirely superseded by `pay-sui` or `pay-all-sui` depending on whether an amount is specified or not.

Author: Thanks, updated!

╰───────────────────┴────────────────────────────────────────────────────────────────────┴────────────────╯
````

Address `hungry-spodumene` has a few coins:

```
sui client gas hungry-spodumene
╭────────────────────────────────────────────────────────────────────┬────────────────────┬──────────────────╮
│ gasCoinId                                                          │ mistBalance (MIST) │ suiBalance (SUI) │
├────────────────────────────────────────────────────────────────────┼────────────────────┼──────────────────┤
│ 0x205972830acd8264a38b9a1776a1b72a1ae626fa23a4d1f12249349ce3b83c06 │ 200000000000       │ 200.00           │
│ 0x56d76420a5bc7d356e3930e6a2ddc61cbbc0c87ea5c5f3cc3ac5952b4d52be5a │ 200000000000       │ 200.00           │
│ 0xc9b447fff18f13fa035e028534b8344d5fc8a8760248fad10155e78f44dc3a52 │ 200000000000       │ 200.00           │
│ 0xe373de9bdbd3dbd4f5f707f144c83af74a181cdb3903a953ee366f48e33865c7 │ 200000000000       │ 200.00           │
│ 0xf7adb6aeb18eb409c03fe88cc31711b61a65333e0fdd88c1815d4ec75f713f31 │ 200000000000       │ 200.00           │
╰────────────────────────────────────────────────────────────────────┴────────────────────┴──────────────────╯
```

> We want to send 0.5 SUI to `eloquent-amber`. Given that we have a few gas coins, we can use Pay. If we would have had only one gas coin, then we could have only used TransferSui, or PaySui, or we would have needed to split the coin first to have another coin to use for paying gas. In this case, let's use the TransferSui command as we will not need to provide a gas coin to be used for the gas fees. We need to pass the address (or its alias) of the address to which SUI is sent, which coin to use to transfer SUI from, and the amount of SUI to transfer.

Reviewer 2: ```suggestion
You want to send 0.5 SUI to `eloquent-amber`. Given that you have a few gas coins, you can use `pay`. If you had only one gas coin, then you would need to use `transfer-sui` or `pay-sui`, or you would need to split the coin to have another coin available for paying gas. In this case, let's use the `transfer-sui` command as you don't need to provide a gas coin for the gas fees. You need to pass the address (or its alias) to receive SUI, the coin to transfer SUI from, and the amount of SUI to transfer.
```

│  ┌──                                                                                              │
│  │ Owner: Account Address ( 0x0fe375fff0ee40d20c54a7f2478b9b5c7eaa3625b7611f9661ec5faefb4a6fea )  │
│  │ CoinType: 0x2::sui::SUI                                                                        │
│  │ Amount: -501997880                                                                             │
│  └──                                                                                              │
│  ┌──                                                                                              │
│  │ Owner: Account Address ( 0x3d78abc33ccd173c521b4f9e78b21ea2c989960b631732c539efaa38bda30976 )  │
│  │ CoinType: 0x2::sui::SUI                                                                        │
│  │ Amount: 500000000                                                                              │
│  └──                                                                                              │
╰───────────────────────────────────────────────────────────────────────────────────────────────────╯

```

> We can check the gas for the sender and the recipient's coins using `sui client gas <address>`. The sender now has 199.49 SUI for that gas coin that was used. 0.5 SUI was transferred, and the rest 0.01 SUI were the gas fees.

Reviewer 2: ```suggestion
You can check the gas for the sender and the recipient's coins using `sui client gas <address>`. The sender now has 199.49 SUI for that gas coin that was used. 0.5 SUI was transferred, and the remaining 0.01 SUI paid the gas fees.
```

│ 0x56d76420a5bc7d356e3930e6a2ddc61cbbc0c87ea5c5f3cc3ac5952b4d52be5a │ 200000000000       │ 200.00           │
│ 0xc9b447fff18f13fa035e028534b8344d5fc8a8760248fad10155e78f44dc3a52 │ 199498002120       │ 199.49           │
│ 0xe373de9bdbd3dbd4f5f707f144c83af74a181cdb3903a953ee366f48e33865c7 │ 200000000000       │ 200.00           │
│ 0xf7adb6aeb18eb409c03fe88cc31711b61a65333e0fdd88c1815d4ec75f713f31 │ 200000000000       │ 200.00           │
╰────────────────────────────────────────────────────────────────────┴────────────────────┴──────────────────╯

sui client gas eloquent-amber
╭────────────────────────────────────────────────────────────────────┬────────────────────┬──────────────────╮
│ gasCoinId                                                          │ mistBalance (MIST) │ suiBalance (SUI) │
├────────────────────────────────────────────────────────────────────┼────────────────────┼──────────────────┤
│ 0xa031d3f8fd53b5f9885172e1c6bd8b770b25e42371c9ee215c6d4b21b2b73241 │ 500000000          │ 0.50             │
╰────────────────────────────────────────────────────────────────────┴────────────────────┴──────────────────╯
```

> If you'd like to send coins to multiple addresses, use one of the `Pay*` commands.

Reviewer 2: ```suggestion
If you'd like to send coins to multiple addresses, use one of the `pay*` commands.
```

> If you'd like to transfer the whole object, you can use `sui client transfer-sui` and do not pass an amount:

Reviewer 2: ```suggestion
If you want to transfer the whole object, use `sui client transfer-sui` and do not pass an amount:
```

```
sui client transfer-sui --to eloquent-amber --sui-coin-object-id 0xc9b447fff18f13fa035e028534b8344d5fc8a8760248fad10155e78f44dc3a52 --gas-budget 5000000
```

> Then check again the gas for `eloquent-amber`

Reviewer 2: ```suggestion
Then check the gas for `eloquent-amber` again:
```
